File: go/problems/day3_test.go
package problems

import "testing"

var day3 Solvable[int] = Day3{}

func TestDay3_TestSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day3_1")
	result := day3.Solve(input)

	expected := 4361
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

func TestDay3_FullSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day3_2")
	result := day3.Solve(input)

	expected := 533784
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

File: ts/problems/day3.test.ts
import test, { describe } from "node:test";
import assert from "node:assert";

import solve from "./day3.js";
import { readProblemInput } from "../inputReader.js";

describe("Day 3", () => {
  test("test set should yield expected result", () => {
    const input = readProblemInput("day3_1");
    const result = solve(input);
    assert.equal(result, 4361);
  });

  test("full set should yield expected result", () => {
    const input = readProblemInput("day3_2");
    const result = solve(input);
    assert.equal(result, 533784);
  });
});
